File: test/for.py
@transpile
def rangeStop():
    stop = 10
    total = 0
    for i in range(stop):
        total += i*2
@transpile
def rangeStartStop():
    start = 5
    stop = 15
    product = 1
    for i in range(start, stop):
        if i % 2 == 0:
            product *= i
@transpile
def rangeStartStopStep():
    start = 2
    stop = 20
    step = 4
    weighted_sum = 0
    index = 0
    for i in range(start, stop, step):
        weighted_sum += i * index
        index += 1

File: test/for.cpp
#include <ostream>
void rangeStop();

void rangeStartStop();

void rangeStartStopStep();

void rangeStop()
{
  int stop = 10;
  int total = 0;
  for (int i = 0; i < stop; ++i) {
    total += i * 2;
  }
}

void rangeStartStop()
{
  int start = 5;
  int stop = 15;
  int product = 1;
  for (int i = start; i < stop; ++i) {
    if (i % 2 == 0) {
      product *= i;
    }
  }
}

void rangeStartStopStep()
{
  int start = 2;
  int stop = 20;
  int step = 4;
  int weighted_sum = 0;
  int index = 0;
  for (int i = start; i < stop; i += step) {
    weighted_sum += i * index;
    index += 1;
  }
}

